src/components/HOC/withApi.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+const withApi = (WrappedComponent, charactersQuantity) => {
+    class WithApi extends React.Component {
+        constructor(props) {
+            super(props)
+            this.state = {
+                charactersCount: 0,
+                charactersArr: [],
+                randomIds: [],
+                test: "test works"
+            }
+        }
+        
+        getCharactersArr = (callback) =>{
+            this.getCharactersCount(() => {
+                this.getRandomIds(charactersQuantity, () => {
+                    this.getCharacters(callback);
```

Comment:
Beautiful example of the callback hell :)
Only 3 level, but still  its very unreadable!
try to refactor this code so that it uses `async` / `await` and I ensure you that it will be much more readable :)

Reply:
Still do not know how to slove it since setState is async, but do not returns promise...